subcommand/server-acl-init/command.go

```diff
@@ -743,45 +752,51 @@ func (c *Command) consulDatacenter(client *api.Client) (string, error) {
 			return opErr
 		})
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	if _, ok := agentCfg["Config"]; !ok {
-		return "", fmt.Errorf("/agent/self response did not contain Config key: %s", agentCfg)
+		return "", "", fmt.Errorf("/agent/self response did not contain Config key: %s", agentCfg)
 	}
 	if _, ok := agentCfg["Config"]["Datacenter"]; !ok {
-		return "", fmt.Errorf("/agent/self response did not contain Config.Datacenter key: %s", agentCfg)
+		return "", "", fmt.Errorf("/agent/self response did not contain Config.Datacenter key: %s", agentCfg)
 	}
 	dc, ok := agentCfg["Config"]["Datacenter"].(string)
 	if !ok {
-		return "", fmt.Errorf("could not cast Config.Datacenter as string: %s", agentCfg)
+		return "", "", fmt.Errorf("could not cast Config.Datacenter as string: %s", agentCfg)
 	}
 	if dc == "" {
-		return "", fmt.Errorf("value of Config.Datacenter was empty string: %s", agentCfg)
+		return "", "", fmt.Errorf("value of Config.Datacenter was empty string: %s", agentCfg)
+	}
+	primaryDC, ok := agentCfg["DebugConfig"]["PrimaryDatacenter"].(string)
```

Comment:
In the future we will have `agentCfg["Config"]["PrimaryDatacenter"]` but since it doesnt exist going to fall back on the `DebugConfig` and address it again in the future when we have an overlapping release with consul.

Reply:
Can we swap this over with the consul `1.10-rc` release?

Reply:
It's not in yet afaik, not sure when it will be in tbh.
But it definitely will not work with older consul versions at that point. We could do some sort of attempt on both Debug/Config which is doable.

Reply:
This should try `Config` and if it's not there, fall back to `DebugConfig`. Then we'll be safe if it's removed from `DebugConfig`.